common-graphql/src/main/scala/explore/common/SequenceStepsGQL.scala

```diff
@@ -0,0 +1,350 @@
+                scienceS:science {
+                  ...southSequenceFields
+                }
+              }
+            }
+          }
+        }
+      }
+
+      fragment northSequenceFields on GmosNorthSequence {
+        atoms {
+          id
+          steps {
+            id
+            stepType
+            instrumentConfig {
+              exposure {
+                microseconds
+              }
+              grating {
+                disperser
+                wavelength {
+                  picometers
+                }
+              }
+              fpu {
+                ... on GmosNorthBuiltinFpu {
+                  builtin
+                }
+              }
+              filter
+              readout {
+                xBin
+                yBin
+              }
+              roi                        
+            }
+            stepConfig {
+              __typename
+              ... on Gcal {
+                shutter
+              }                        
+              ... on Science {
+                offset {
+                  p {
+                    microarcseconds
+                  }
+                  q {
+                    microarcseconds
+                  }
+                }
+              }
+            }
+          }
+        }      
+      }
+
+      fragment southSequenceFields on GmosSouthSequence {
+        atoms {
+          id
+          steps {
+            id
+            stepType
+            instrumentConfig {
+              exposure {
+                microseconds
+              }
+              grating {
+                disperser
+                wavelength {
+                  picometers
+                }
+              }
+              fpu {
+                ... on GmosSouthBuiltinFpu {
+                  builtin
+                }
+              }
+              filter
+              readout {
+                xBin
+                yBin
+              }
+              roi                        
+            }
+            stepConfig {
+              __typename
+              ... on Gcal {
+                shutter
+              }                        
+              ... on Science {
+                offset {
+                  p {
+                    microarcseconds
+                  }
+                  q {
+                    microarcseconds
+                  }
+                }
+              }
+            }
+          }
+        }      
+      }
+    """
+
+    implicit def offsetComponentDecoder[T]: Decoder[Offset.Component[T]] = Decoder.instance(c =>
+      c.downField("microarcseconds")
+        .as[Long]
+        .map(Angle.signedMicroarcseconds.reverse.composeIso(Offset.Component.angle[T].reverse).get)
+    )
+
+    implicit val offsetDecoder: Decoder[Offset] = Decoder.instance(c =>
+      for {
+        p <- c.downField("p").as[Offset.P]
+        q <- c.downField("q").as[Offset.Q]
+      } yield Offset(p, q)
+    )
+
+    sealed trait SeqSite {
```

Comment:
It took a bit of work to get a unified model we can work with, since there are enums involved that are different for North and South, and they have no common ancestor type.